Aspire Budgeting/Facilities/FileValidator.swift

```diff
@@ -0,0 +1,147 @@
+//
+// FileValidator.swift
+// Aspire Budgeting
+//
+
+import Combine
+import Foundation
+import GoogleAPIClientForREST
+import GTMSessionFetcher
+
+enum FileValidatorState {
+  case isLoading
+  case dataMapRetrieved([String: String])
+  case error(Error)
+}
+
+enum GoogleSheetsValidationError: String, Error {
+  case noSheetsInSpreadsheet = "No sheets in spreadsheet"
+  case noNamedRangesInSpreadsheet = "No named ranges in spreadsheet"
+  case internalParsingError = "Internal parsing error"
+  case invalidSheet = "Invalid Sheet"
+}
+
+protocol FileValidator {
+  var currentState: CurrentValueSubject<FileValidatorState, Never> { get }
+  func validate(file: File, for: User)
+}
+
+class GoogleSheetsValidator: FileValidator {
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/aspirebudgetingmobile/aspirebudgeting_ios/blob/11e7508ea9c6ba09d42329eff1bf52e705f4298b/Aspire Budgeting/Facilities/FileValidator.swift#L29"></span>Consider using final for this class or use a struct (final_class)</td>
    </tr>
  </tbody>
</table>